### DISCUSSION TITLE
no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"

### DISCUSSION THREAD

**Discussion by jhall-bondi** [onepanelio/core#980] [Get Help]
Hi

Following the quick start guide setting up the project in WSL. Came across this error after using the "opctl apply" which has me stumped.
output below.
`$ opctl apply
Starting deployment...

clusterrole.rbac.authorization.k8s.io/application-manager-role configured
clusterrolebinding.rbac.authorization.k8s.io/application-manager-rolebinding configured
namespace/application-system unchanged
service/application-controller-manager-service unchanged
statefulset.apps/application-controller-manager unchanged

Failed: unable to recognize ".onepanel/application.kubernetes.yaml": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"`

Some googling found this issue.  [https://stackoverflow.com/questions/69054622/unable-to-install-crds-in-kubernetes-kind](url) 
So either I have not followed a step correctly, or maybe the manifest is applying outdated API versions?

Some pointers in the right direction would be appreciated.
Josh.

**Comment by ghost** [comment]
Could it be possible that you are running server version of 1.22 or higher?

My server version is 1.21 and it runs correctly. (I am using Azure Kubernetes Services) 
My client version is 1.22.

This can show you which version you are using:

```
kubectl version
```

**Reply by jhall-bondi** [reply]
Thanks for the tip @nickviumore. Managed to resolve this.

I created a new cluster but added the --version 1.21 into the command which forces the cluster to install 1.21 kubernetes server. 
'eksctl create cluster --name=<name> --version 1.21 --region <region> --zones <region>a,<region>b --node-zones <region>a --nodes 2 --node-type m5.xlarge --node-volume-size 100 --nodes-min 2 --nodes-max 5 --asg-access --ssh-access --managed=false --tags "onepanel.io/enabled=true,k8s.io/cluster-autoscaler/node-template/label/node.kubernetes.io/instance-type=m5.xlarge"'

So my server is version 1.21
My client is version 1.21

I assume anyone following the guide who leaves out the --version 1.21 parameter in eksctl create cluster may run into this issue as it will default to server version 1.22 which deprecates some things.

**Comment by rikhenderickx** [comment]
Hello,

I ran into the exact same issue when deploying OnePanel in Azure (using Azure Kubernetes Services).
Version 1.21 is no longer supported by Azure so the workaround described above doesn't do the trick.

Is there any other way to resolve this?
As @jhall-bondi mentioned: anyone following the guide (using Azure) may be running into this issue now too.

Thanks!
Rik
